torch/optim/lr_scheduler.py

```diff
@@ -231,6 +231,8 @@ def get_lr(self):
             warnings.warn("To get the last learning rate computed by the scheduler, "
                           "please use `get_last_lr()`.")
 
+        if self.last_epoch == 0:
+            return self.base_lrs
```

Comment:
This is BC-breaking. Do you have a sample code that gives unexpected results due to this?

Reply:
``` python
import torch
import torch.nn as nn
import torch.optim as optim

net = nn.Linear(2, 3)
optimizer = optim.SGD(net.parameters(), 0.1)
print("before lr_scheduler init:", [group['lr'] for group in optimizer.param_groups])

lr_scheduler = optim.lr_scheduler.LambdaLR(optimizer, lambda epoch:0.5)
print("after lr_scheduler init:", [group['lr'] for group in optimizer.param_groups])
```
``` bash
before lr_scheduler init: [0.1]
after lr_scheduler init: [0.05]
```

learning rate changes immediately after lr_scheduler initialized. I'm not sure this is the desired behavior.